openquake/hazardlib/calc/gmf.py

```diff
@@ -244,18 +237,7 @@ def ground_motion_fields(rupture, sites, imts, gsim, truncation_level,
         for all sites in the collection. First dimension represents
         sites and second one is for realizations.
     """
-    r_sites = rupture_site_filter.affected(rupture, sites)
-    if r_sites is None:
-        return dict((imt, numpy.zeros((len(sites), realizations)))
-                    for imt in imts)
-    gc = GmfComputer(rupture, r_sites, [str(imt) for imt in imts], [gsim],
+    gc = GmfComputer(rupture, sites, [str(imt) for imt in imts], [gsim],
                      truncation_level, correlation_model)
     res = gc.compute(gsim, realizations, seed)
-    result = {}
-    for imti, imt in enumerate(gc.imts):
-        # makes sure the lenght of the arrays in output is the same as sites
-        if rupture_site_filter is not filters.rupture_site_noop_filter:
-            result[imt] = r_sites.expand(res[imti], placeholder=0)
-        else:
-            result[imt] = res[imti]
-    return result
+    return {imt: res[imti] for imti, imt in enumerate(gc.imts)}
```

Comment:
Is this change implying that the sites received by the GMF calculator have been already filtered?